Simplified/MyBooks/NYPLMyBooksDownloadCenter.m

```diff
@@ -1620,5 +1638,56 @@ - (void)downloadProgressDidUpdateTo:(double)progress forBook:(NYPLBook * _Nonnul
 }
 #endif
 
+#if FEATURE_AUDIOBOOKS
+/// Create an audiobook object and add to download queue
+- (void)downloadAudiobookForBook:(NYPLBook *)book {
```

Comment:
would it be possible to have this in a swift extension? For large methods usually that's the preference, instead of adding more objc code. But I also know there are issues sometimes with that. Just wondering.

Reply:
If we do it in Swift extension, we need to expose the `broadcastUpdate` function and the `audiobookDownloader` variable. I was trying to avoid that when writing this part of code, but I can do that if you think it is a good trade off.

Reply:
ah, I see what you mean.... yeah I do think it would be a good tradeoff, because objc is just harder to deal with, and exposing a couple more things is not a big deal (audiobookDownloader could be readonly, for instance). 

However, you already wrote it and tested in objc, so I feel bad.... it can be a separate PR and perhaps a separate ticket maybe?

Reply:
Sure, rewriting this in Swift should not take long, I will include it in this PR.